spec/models/enrollment_action/simple_renewal_spec.rb

```diff
@@ -0,0 +1,198 @@
+
+  let(:plan) { instance_double(Plan, :id => 1) }
+  let(:policy) { instance_double(Policy, :enrollees => [enrollee_primary, enrollee_new], :eg_id => 1) }
+
+  let(:action_event) { instance_double(
+    ::ExternalEvents::EnrollmentEventNotification,
+    :existing_policy => policy,
+    :event_xml => event_xml,
+    :all_member_ids => [1,2],
+    :event_responder => event_responder,
+    :hbx_enrollment_id => 1,
+    :employer_hbx_id => 1
+  ) }
+
+  let(:termination_helper_result_xml) { double }
+  let(:termination_writer_result_xml) { double }
+
+  let(:termination_publish_helper) { instance_double(
+    EnrollmentAction::ActionPublishHelper,
+    :to_xml => termination_helper_result_xml
+  ) }
+
+  let(:action_helper_result_xml) { double }
+
+  let(:action_publish_helper) { instance_double(
+    EnrollmentAction::ActionPublishHelper,
+    :to_xml => action_helper_result_xml
+  ) }
+
+  let(:termination_writer) {
+    instance_double(::EnrollmentAction::EnrollmentTerminationEventWriter)
+  }
+
+  let(:employer_hbx_id) { double }
+  let(:terminated_policy_eg_id) { double }
+  let(:employer) { instance_double(Employer, :hbx_id => employer_hbx_id) }
+
+  let(:terminated_policy) {
+    instance_double(Policy, :eg_id => terminated_policy_eg_id, :employer => employer)
+  }
+
+  subject do
+    EnrollmentAction::SimpleRenewal.new(nil, action_event).tap do |s|
+      s.terminated_policy_information = [[terminated_policy, [1,2]]]
+    end
+  end
+
+  before :each do
+    allow(::EnrollmentAction::EnrollmentTerminationEventWriter).to receive(:new).with(terminated_policy, [1,2]).and_return(termination_writer)
+    allow(termination_writer).to receive(:write).with("transaction_id_placeholder", "urn:openhbx:terms:v1:enrollment#terminate_enrollment").and_return(termination_writer_result_xml)
+    allow(::EnrollmentAction::ActionPublishHelper).to receive(:new).with(event_xml).and_return(action_publish_helper)
+    allow(::EnrollmentAction::ActionPublishHelper).to receive(:new).with(termination_writer_result_xml).and_return(termination_publish_helper)
+    allow(subject).to receive(:publish_edi).with(amqp_connection, termination_helper_result_xml, terminated_policy_eg_id, employer_hbx_id)
+    allow(subject).to receive(:publish_edi).with(amqp_connection, action_helper_result_xml, action_event.hbx_enrollment_id, action_event.employer_hbx_id)
+    allow(action_publish_helper).to receive(:set_event_action).with("urn:openhbx:terms:v1:enrollment#auto_renew")
+    allow(action_publish_helper).to receive(:keep_member_ends).with([])
+  end
+
+  it "publishes termination resulting xml to edi" do
+    expect(subject).to receive(:publish_edi).with(amqp_connection, termination_helper_result_xml, terminated_policy_eg_id, employer_hbx_id)
+    subject.publish
+  end
+
+  it "publishes an event of enrollment auto renewing" do
+    expect(action_publish_helper).to receive(:set_event_action).with("urn:openhbx:terms:v1:enrollment#auto_renew")
```

Comment:
This is not the correct event type.  It should be "initial".

Reply:
please do review it changed event type from auto_renew to active_renew